test/integration/apiary.js

```diff
@@ -0,0 +1,199 @@
+
+describe("tests the /liquidity_pools endpoint", function() {
+  const lpId = "0569b19c75d7ecadce50501fffad6fe8ba4652455df9e1cc96dc408141124dd5";
+
+  it("GET /", function(done) {
+    let server = new StellarSdk.Server(MOCK_SERVER, {allowHttp: true});
+
+    chai.request(MOCK_SERVER)
+      .get("/liquidity_pools")
+      .end(function(err, res) {
+        if (err != null) done(err);
+        expect(res.body).not.to.be.null;
+
+        server
+          .liquidityPools()
+          .call()
+          .then(resp => {
+            expect(resp.records).to.deep.equal(res.body._embedded.records);
+            done();
+          })
+          .catch(err => done(err));
+      });
+  });
+
+  it('GET /<id>', function(done) {
+    let server = new StellarSdk.Server(MOCK_SERVER, {allowHttp: true});
+
+    chai.request(MOCK_SERVER)
+      .get(`/liquidity_pools/${lpId}`)
+      .end(function(err, res) {
+        if (err != null) done(err);
+        expect(res.body).not.to.be.null;
+
+        server
+          .liquidityPools()
+          .liquidityPoolId(lpId)
+          .call()
+          .then(resp => {
+            expect(resp).to.deep.equal(res.body);
+            done();
+          })
+          .catch(err => done(err));
+      });
+  });
+
+  let server = new StellarSdk.Server(MOCK_SERVER, {allowHttp: true});
+  const testCases = {
+    transactions: server.transactions(),
+    operations: server.operations(),
+    effects: server.effects(),
+  };
+
+  Object.keys(testCases).forEach(suffix => {
+    it(`GET /<id>/${suffix}`, function(done) {
+      chai.request(MOCK_SERVER)
+        .get(`/liquidity_pools/${lpId}/${suffix}`)
+        .end(function(err, res) {
+          if (err != null) return done(err);
+          expect(res.body).not.to.be.null;
+
+          testCases[suffix]
+            .forLiquidityPool(lpId)
+            .call()
+            .then(resp => {
+              resp.records.forEach((record, i) => {
+                // In an ideal world, we'd do a `.deep.equal(...)` here against
+                // the server response. However, these methods provide
+                // convenience methods that cause them to not perfectly align
+                // with the server response.
+                //
+                // For example, transaction responses include the ledger number,
+                // but this is attached to the `ledger_attr` field in a
+                // `TransactionRecord`, while `ledger` is a function that
+                // corresponds to a `LedgerCallBuilder`.
+                //
+                // For this reason, we do a "best effort" validity check by at
+                // least ensuring that the response objects have *at least* all
+                // of the keys in the server response.
+                expect(isSubsetOf(
+                  Object.keys(record),
+                  Object.keys(res.body._embedded.records[i])
+                )).to.be.true;
```

Comment:
Thanks for the lodash reference! Because the values won't always be equal due to the fact that responses aren't a 1-to-1 mapping. For example, `TransactionRecord.ledger` -> a `LedgerCallBuilder`, while the JSON's `.ledger` is an integer. This is only true for transactions, though, so I've updated the tests to work around that in ce659298.